tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,419 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	v1beta1 "github.com/NVIDIA/k8s-dra-driver-gpu/pkg/nvidia.com/clientset/versioned"
+	"github.com/NVIDIA/k8s-test-infra/pkg/diagnostics"
+	"helm.sh/helm/v3/pkg/repo"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	helm "github.com/mittwald/go-helm-client"
+	helmValues "github.com/mittwald/go-helm-client/values"
+	corev1 "k8s.io/api/core/v1"
+	extclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	// DefaultNamespaceDeletionTimeout is timeout duration for waiting for a namespace deletion.
+	DefaultNamespaceDeletionTimeout = 10 * time.Minute
+
+	// PollInterval is how often to Poll pods, nodes and claims.
+	PollInterval = 2 * time.Second
+)
+
+// Test context
+var (
+	Kubeconfig         string
+	HelmChart          string
+	HelmValuesFile     string
+	HelmValues         string
+	HostManagedDrivers bool
+
+	// Image
+	ImageRepo       string
+	ImageTag        string
+	ImagePullPolicy string
+
+	// k8s clients
+	clientConfig   *rest.Config
+	clientSet      clientset.Interface
+	extClient      *extclient.Clientset
+	resourceClient *v1beta1.Clientset
+
+	testNamespace *corev1.Namespace // Every test has at least one namespace unless creation is skipped
+	EnableGFD     bool
+
+	// Helm
+	helmClient      helm.Client
+	helmLogFile     *os.File
+	helmArtifactDir string
+	helmLogger      *log.Logger
+	helmReleaseName string
+	helmOptions     helmValues.Options
+
+	ctx     context.Context
+	Timeout time.Duration
+	cwd     string
+
+	diagnosticsCollector *diagnostics.Diagnostic
+	CollectLogsFrom      []string
+	LogArtifactDir       string
+	packagePath          string
+)
+
+func TestMain(t *testing.T) {
+	suiteName := "E2E K8s DRA GPU Driver"
+
+	RegisterFailHandler(Fail)
+
+	ctx = context.Background()
+	getTestEnv()
+
+	RunSpecs(t,
+		suiteName,
+	)
+}
+
+// BeforeSuite runs before the test suite
+var _ = BeforeSuite(func() {
+	var err error
+
+	cwd, err = os.Getwd()
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get k8s clients
+	getK8sClients()
+
+	// Create clients for apiextensions and our CRD api
+	extClient = extclient.NewForConfigOrDie(clientConfig)
+
+	// Create a nvidia.com resource client
+	resourceClient, err = v1beta1.NewForConfig(clientConfig)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Create a namespace for the test
+	testNamespace, err = CreateTestingNS("nvidia-dra-driver-gpu", clientSet, nil)
```

Comment:
That is the namespace that we recommend to install the DRA driver in, right?

The normal user workflow has workloads running in a different namespace, right?

Reply:
>That is the namespace that we recommend to install the DRA driver in, right?

No, this will be a namesapce that will look like `nvidia-dra-driver-gpu-<random>`  just for the sake of the test run.

>The normal user workflow has workloads running in a different namespace, right?
Yes, a recommended name for the namespace could be `nvidia-dra-driver-gpu` but that is really up to the user.

Reply:
> No, this will be a namesapce that will look like nvidia-dra-driver-gpu-<random> just for the sake of the test run.

Ah, nice! I didn't look at the implementation. So, the arg is just a prefix. Maybe use the word "test" in there?